ktor-server/ktor-server-core/common/src/io/ktor/server/http/HttpRequestLifecycle.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2014-2025 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.http
+
+import io.ktor.server.application.*
+import io.ktor.server.application.hooks.*
+import io.ktor.util.*
+import io.ktor.utils.io.*
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.cancel
+
+/**
+ * Configuration for the [HttpRequestLifecycle] plugin.
+ */
+public class HttpRequestLifecycleConfig internal constructor() {
+    /**
+     * When `true`, cancels the call coroutine context if the other peer resets the client connection.
+     * When `false` (default), request processing continues even if the connection is closed.
+     *
+     * Example:
+     * ```kotlin
+     * install(HttpRequestLifecycle) {
+     *     cancelCallOnClose = true
+     * }
+     * ```
+     */
```

Comment:
It's not clear from the documentation where the cancellation exception will be thrown. Could you please add an example of handling the exception as well?

Reply:
It is also unclear what the use case for the new property is - when should someone use it?